src/main/java/com/powsybl/openloadflow/network/impl/Transformers.java

```diff
@@ -27,26 +27,38 @@ private Transformers() {
      * Get ratio on side 1 of a two windings transformer.
      */
     public static double getRatio(TwoWindingsTransformer twt) {
+        return getRatio(twt, getCurrentPosition(twt.getRatioTapChanger()), getCurrentPosition(twt.getPhaseTapChanger()));
+    }
+
+    public static double getRatio(TwoWindingsTransformer twt, int rtcPosition, int ptcPosition) {
         double rho = twt.getRatedU2() / twt.getRatedU1();
         if (twt.getRatioTapChanger() != null) {
-            rho *= twt.getRatioTapChanger().getCurrentStep().getRho();
+            rho *= twt.getRatioTapChanger().getStep(rtcPosition).getRho();
         }
         if (twt.getPhaseTapChanger() != null) {
-            rho *= twt.getPhaseTapChanger().getCurrentStep().getRho();
+            rho *= twt.getPhaseTapChanger().getStep(ptcPosition).getRho();
         }
         return rho;
     }
 
+    private static int getCurrentPosition(RatioTapChanger rtc) {
+        return rtc != null ? rtc.getTapPosition() : -1;
```

Comment:
The problem with this design is that -1 could be a valid tap position

Reply:
So what else should I do ?

Reply:
Indeed, this value "-1" is never used in the code because we have some checks after:
````
public static double getRatioLeg(ThreeWindingsTransformer twt, ThreeWindingsTransformer.Leg leg, int rtcPosition, int ptcPosition) {
        double rho = twt.getRatedU0() / leg.getRatedU();
        if (leg.getRatioTapChanger() != null) {
            rho *= leg.getRatioTapChanger().getStep(rtcPosition).getRho();
        }
        if (leg.getPhaseTapChanger() != null) {
            rho *= leg.getPhaseTapChanger().getStep(ptcPosition).getRho();
        }
        return rho;
    }
````
but I need to return an integer in my function, even if I don't have any RTC or PTC. Do you have an other suggestion ?